### PR TITLE
[Post] New API Blueprint Tutorial and Examples

### PR DESCRIPTION
**Pull request by kylef**
This pull request introduces a new post on the new API Blueprint tutorial, examples and implementation along with demonstrating some of the features of this API implementation.

## :warning: DO NOT MERGE: Please just review :warning:

### CODE REVIEW

File: _posts/2015-03-03-Polls-API-Blueprint-Tutorial-Examples.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+}
+```
+
+We've deployed an instance of this API and it's accessible for anyone to experiment with: `https://polls.apiblueprint.org/`.
+
+You can also deploy your own instance of the Polls API to Heroku with a single click:
+
+[![Deploy to Heroku](https://www.herokucdn.com/deploy/button.png)](https://heroku.com/deploy?template=https://github.com/apiaryio/polls-api)
+
+The Polls API is also a [hypermedia](TODO) API supporting the [HAL](http://stateless.co/hal_specification.html) and [Siren](https://github.com/kevinswiber/siren) hypermedia content-types based on [content negotiation](https://github.com/apiaryio/api-blueprint/blob/master/examples/6.%20Requests.md). This is demonstrated in an advanced version of the Polls API [example blueprint](http://docs.pollshypermedia.apiary.io/).
```
Comment (kylef) [inline _posts/2015-03-03-Polls-API-Blueprint-Tutorial-Examples.md]: I'd like to point readers to beginners information on what hypermedia in the hypermedia hyperlink. Does anyone have any suggestions of a decent resource?

/c @smizell

Comment (fosrias) [inline _posts/2015-03-03-Polls-API-Blueprint-Tutorial-Examples.md]: > I'd like to point readers to beginners information on what hypermedia in
the hypermedia hyperlink.

@kylef I don't understand this question. Are you saying you have an empty
hyperlink you want to connect to something?

Comment (fosrias) [inline _posts/2015-03-03-Polls-API-Blueprint-Tutorial-Examples.md]: Also, this is a really great introduction:
http://oredev.org/2010/sessions/hypermedia-apis

Also, you might look at:

http://www.infoq.com/articles/hypermedia-api-tutorial-part-one
http://www.infoq.com/articles/hypermedia-api-tutorial-part-two

Comment (kylef) [inline _posts/2015-03-03-Polls-API-Blueprint-Tutorial-Examples.md]: Yes, that's exactly what I was saying. To go in the above line (`[hypermedia](TODO)`).

Comment (fosrias) [inline _posts/2015-03-03-Polls-API-Blueprint-Tutorial-Examples.md]: @kylef Sent you some references. If you want something else, I have some
more possibilities.

Comment (kylef) [inline _posts/2015-03-03-Polls-API-Blueprint-Tutorial-Examples.md]: Great, thanks @fosrias I'll take a look though them!

File: _posts/2015-03-03-Polls-API-Blueprint-Tutorial-Examples.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+title: New API Blueprint Tutorial and Examples
+excerpt: We've introduced a brand new API Blueprint tutorial along with some new examples.
+layout: post
+date: 2015-03-03 16:00:00 +0000
+author: kyle
+published: true
+comments: true
+---
+
+We've introduced a new [API Blueprint tutorial](http://apiary.io/blueprint) along with some brand new [API Blueprint examples](http://docs.pollsapi.apiary.io/). Our new tutorial walks you though the basics of API Blueprint along with how you can build your very first blueprint. It uses an example of a polls service, which allows consumers to view polls and vote in them.
```
Comment (zdne) [inline _posts/2015-03-03-Polls-API-Blueprint-Tutorial-Examples.md]: Since this blog post is introducing it I would say it shouldn't start with this sentence .. :)

File: _posts/2015-03-03-Polls-API-Blueprint-Tutorial-Examples.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        }, {
+            "choice": "Python",
+            "url": "/questions/1/choices/2",
+            "votes": 0
+        }
+    ]
+}
+```
+
+We've deployed an instance of this API and it's accessible for anyone to experiment with: `https://polls.apiblueprint.org/`.
```
Comment (zdne) [inline _posts/2015-03-03-Polls-API-Blueprint-Tutorial-Examples.md]: I would just skip the "We've deployed an instance of this API and" completely

File: _posts/2015-03-03-Polls-API-Blueprint-Tutorial-Examples.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+You can also deploy your own instance of the Polls API to Heroku with a single click:
+
+[![Deploy to Heroku](https://www.herokucdn.com/deploy/button.png)](https://heroku.com/deploy?template=https://github.com/apiaryio/polls-api)
+
+The Polls API is also a [hypermedia](http://www.infoq.com/articles/hypermedia-api-tutorial-part-one) API supporting the [HAL](http://stateless.co/hal_specification.html) and [Siren](https://github.com/kevinswiber/siren) hypermedia content-types based on [content negotiation](https://github.com/apiaryio/api-blueprint/blob/master/examples/6.%20Requests.md). This is demonstrated in an advanced version of the Polls API [example blueprint](http://docs.pollshypermedia.apiary.io/).
```
Comment (zdne) [inline _posts/2015-03-03-Polls-API-Blueprint-Tutorial-Examples.md]: Seems the http://docs.pollshypermedia.apiary.io is private?

**Comment by zdne** [comment]
I think this needs a bit more work. Similarly to Siren / HAL sections I would add (before these two) a JSON API section that would introduce the JSON (basic) version. 

In addition I would expect a mention of the default blueprint and the benefits of having the host there with some demonstration of the Apiary Console and stuff. 

Also are you planning to do a HTML version of it. Similarly I believe the API implementation https://github.com/apiaryio/polls-api should be tested with Dredd and that should be pointed out in the article. 

In a nutshell it may be a good way to think about this as a Tutorial for Apiary / API Blueprint ecosystem..

**Comment by smizell** [comment]
Just passing by.

> I would add (before these two) a JSON API section that would introduce the JSON (basic) version.

Just a note–might not want to call it JSON API, as that's [another media type](http://jsonapi.org/) with that title.

If you do HTML (which would be awesome), you could use the [`alternate` link relation](http://www.w3.org/TR/html5/links.html#rel-alternate) to indicate other media types.

**Comment by zdne** [comment]
I will propose a ne PR instead of this one soon. Closing.
